### ISSUE TITLE
Tasker integration

### ISSUE THREAD

**Issue by sik0vny** [gotify/android#44]
Tasker is a powerful scripting system with unlimited possibilites. Integrating it with Gotify, will add another dimension to it:)
Tasker would react on Gotify's messages from defined apps or on a keywords/regex found in messages. After that, Tasker would run a user's defined script.
F.ex: 
- send http request to a server
- play alarm sound
- write sms
- download a file from link in the message

Here is link to the documentation:
https://tasker.joaoapps.com/plugins.html

**Comment by eternal-flame-AD** [comment]
Why not simply let tasker match the text in the notification and take corresponding actions?:) I think that is basically the same as writing a dedicated custom Tasker Event

Plus there is another one reason I am not very in favor of implementing a Tasker Plugin. This is relatively personal but I still wanted to share it with you and the project maintainers. It is in some way promoting a paid and closed-source service(Tasker is a paid app). Gotify relieved me from expensive paid and hosted services such as Pushbullet. I really wish to keep it simple and open so I am not very in hope of adding this extension.

**Comment by sik0vny** [comment]
@dugite-code i agree. That sounds like a good compromise and will not be dependent on any particular application (so: no flame wars). 

One of the use case for this can be that for notifications from certain app/token, user will turn off visible notifications, and just pass-through Intents. And inside those intents some more structured data can be transmitted, f. ex JSON or XML file. This can be used then to feed some more complicated processes in other applications.

**Comment by eternal-flame-AD** [comment]
Thanks for your ideas. I think this would definitely supercharge the power of gotify. I am now voting up for this feature :+1: 

Firstly, I assume that the feature we are asking in this issue is no longer about simply connecting gotify messages with automation apps, but to enable gotify android client to take customed actions on messages(f.ex. no notification; fire an intent; add a button in the notification;), rather than simply firing a plain notification.

Secondly, the feature would need extra data in the message, which requires API changes such as I proposed in https://github.com/gotify/server/issues/120. I think this issue is a dependent on https://github.com/gotify/server/issues/120.

**Comment by sik0vny** [comment]
Thank you @eternal-flame-AD, I'm happy we found a middle ground:) I've seen your other posts/ideas, and I'm fully supporting them.

**Comment by buergi** [comment]
There is a very simple solution for Tasker integration (and probably others as well) which has been available ever since (to be precise since 2018, ee1a841) which requires no plugins or other integrations to work. I wonder that nobody mentioned it. The idea is to catch the broadcast intent sent by gotify-android when it receives a message, [WebSocketService.java:219](https://github.com/gotify/android/blob/211a1b1cba7a364a9d38497432a5b08bbbfe80b8/app/src/main/java/com/github/gotify/service/WebSocketService.java#L219).
As I didn't find any documentation about it as well, I like to briefly explain it here for others to find it:

Create a new profile _Event > System > Intent Received_ and set _Action_ to `com.github.gotify.service.WebSocketService.NEW_MESSAGE`
In the corresponding task you can access the complete message including all extras (as JSON string) in the `%message` variable. You can parse and filter it using simple string operations or even process it e.g. using _AutoTool Json Read_

**Comment by ajl000** [comment]
Is this why people have suggested end-to-end encryption for Gotify.  Does https assist.

**Comment by gohamstergo** [comment]
old thread, but still relevant...

the message from @buergi should really make its way to the Docs as at least a side note. This method works perfectly with Tasker.

**Comment by eternal-flame-AD** [comment]
@gohamstergo Thanks for reminding me on this issue, I have not been following Android part closely (and I am not familiar with Android dev). 

I am certainly willing to accept a PR for this in gotify/website as an unofficial way to do it, because it is not officially supported (I can see in the near future there may be feature requests to make us make this intent filtered) and I have no experience using this so I am not comfortable writing a guide for something I didn't do myself.
